scripts/fabric_links_monitoring.py

```diff
@@ -0,0 +1,183 @@
+                    "COUNTERS_DB", 'COUNTERS:{}'.format(port_oid))
+            port_counters[port_name] = port_counter
+        return port_counters
+
+    def get_ports_error_above_threshold(self, counter_name, threshold):
+        ''' 
+        Return the list of ports which have the counter value above the threshold
+        '''
+        err_ports = []
+        for port_name, port_counter in self.port_counters.items():
+            try:
+                port_counter_value = port_counter[counter_name]
+                prev_port_counter_value = self.prev_port_counters[port_name][counter_name]
+                if (int(port_counter_value) - int(prev_port_counter_value)) > threshold:
+                    err_ports.append(port_name)
+            except KeyError:
+                print('Bad counter_name')
+
+        self.prev_port_counters = self.port_counters.copy()     
+        return err_ports
+    
+    def monitor(self, namespace, error_counter_names, threshold ):
+        err_ports = []
+        
+        for counter_name in error_counter_names:
+            err_ports_per_counter = self.get_ports_error_above_threshold(
+                    counter_name,threshold)
+            if err_ports_per_counter:
+                syslog.syslog(syslog.LOG_CRIT,
+                                  ' {} error above threshold on fabric port {} in {} '.format(counter_name,err_ports_per_counter, namespace))
+                err_ports.extend(err_ports_per_counter)
+        return err_ports
+
+
+class PacketChassisFabricLinkMontoring():
+    '''A class for monitoring internal links in a Packet Chassis.
+
+    Attributes:
+        link_monitor (dict): A dictionary of LinkMonitor objects, keyed by namespace.
+        error_counter_names (list): A list of error counter names to monitor.
+        threshold (int): The threshold for error counters.
+    '''
+
+    def __init__(self):
+        '''Initializes a PacketChassisFabricLinkMontoring object.
+
+        Args:
+            link_monitor (dict): A dictionary of LinkMonitor objects, keyed by namespace.
+            error_counter_names (list): A list of error counter names to monitor.
+            threshold (int): The threshold for error counters.
+
+        Returns:
+            None
+        '''
+
+        self.link_monitor = {}
+        self.appdb = {}
+        self.configdb = {}
+        self.statedb = {}
+        for namespace in self.namespaces:
+            self.appdb[namespace] = self.appl_db(namespace)
+            self.configdb[namespace] = self.config_db(namespace)
+            self.statedb[namespace] = self.state_db(namespace)
+            self.link_monitor[namespace] = FabricLinkErrMontoring(
+                namespace=namespace, port_name_map='COUNTERS_FABRIC_PORT_NAME_MAP',
+                filter_for_up_ports=self.filter_for_up_ports)
+        self.error_counter_names = ['SAI_PORT_STAT_IF_IN_ERRORS']
+        self.threshold = 60
+
+    @property
+    def namespaces(self):
+        namespaces = []
```

Comment:
handle single asic linecard as well?

Reply:
This is monitoring fabric cards from SUP side only and not from LC so need to use namespace.